Code/Framework/AzQtComponents/AzQtComponents/Components/Widgets/CardHeader.cpp

```diff
@@ -186,20 +218,45 @@ namespace AzQtComponents
     void CardHeader::setIcon(const QIcon& icon)
     {
         m_icon = icon;
+        updateIconLabel();
+    }
 
-        if (!icon.isNull())
+    void CardHeader::setIconOverlay(const QIcon& iconOverlay)
+    {
+        m_iconOverlay = iconOverlay;
+        updateIconLabel();
+    }
+
+    void CardHeader::setIconClickable(bool clickable)
+    {
+        m_iconLabel->SetClickable(clickable);
+    }
+
+    bool CardHeader::isIconClickable() const
+    {
+        return m_iconLabel->GetClickable();
+    }
+
+    void CardHeader::updateIconLabel()
+    {
+        if (!m_icon.isNull())
         {
-            m_iconLabel->setPixmap(icon.pixmap(s_iconSize, s_iconSize));
+            QPixmap pixmap = m_icon.pixmap(s_iconSize, s_iconSize);
+            if (!m_iconOverlay.isNull())
+            {
+                QPainter paint(&pixmap);
+                paint.drawPixmap(0, 0, m_iconOverlay.pixmap(s_iconSize, s_iconSize));
+            }
```

Comment:
Nice! This would still work with the above comment about switching to a `QToolButton` instead of a `QLabel` just need to convert the pixmap back to an icon.

Reply:
Good to know! I remembered that I also added the custom clickable QLabel to be able to set a clickable flag on it rather than enabling/disabling the QWidget because the enable/disable property was already being set for visual reasons.